From 44e1240ba89d9ff1050182a001f493735c9f48df Mon Sep 17 00:00:00 2001
From: Ilija Hadzic <ihadzic@research.bell-labs.com>
Date: Sat, 26 May 2012 17:20:22 -0400
Subject: [PATCH 1/3] kms: add RenderNodeMinor option

New RenderNodeMonor option is used to specify the minor
number of the render node. When the option is present, we
try to use the specified render node instead of the physical
card (probed or specified using BusID).

Signed-off-by: Ilija Hadzic <ihadzic@research.bell-labs.com>
---
 src/radeon.h     |    3 ++-
 src/radeon_kms.c |    1 +
 2 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/src/radeon.h b/src/radeon.h
index ce9508c..36405b2 100644
--- a/src/radeon.h
+++ b/src/radeon.h
@@ -227,7 +227,8 @@ typedef enum {
     OPTION_DYNAMIC_PM,
     OPTION_NEW_PLL,
     OPTION_ZAPHOD_HEADS,
-    OPTION_SWAPBUFFERS_WAIT
+    OPTION_SWAPBUFFERS_WAIT,
+    OPTION_RENDER_NODE_MINOR
 } RADEONOpts;
 
 
diff --git a/src/radeon_kms.c b/src/radeon_kms.c
index c094bea..f2a4469 100644
--- a/src/radeon_kms.c
+++ b/src/radeon_kms.c
@@ -73,6 +73,7 @@ const OptionInfoRec RADEONOptions_KMS[] = {
     { OPTION_ZAPHOD_HEADS,   "ZaphodHeads",      OPTV_STRING,  {0}, FALSE },
     { OPTION_PAGE_FLIP,      "EnablePageFlip",   OPTV_BOOLEAN, {0}, FALSE },
     { OPTION_SWAPBUFFERS_WAIT,"SwapbuffersWait", OPTV_BOOLEAN, {0}, FALSE },
+    { OPTION_RENDER_NODE_MINOR, "RenderNodeMinor",  OPTV_INTEGER,  {0}, FALSE },
     { -1,                    NULL,               OPTV_NONE,    {0}, FALSE }
 };
 
-- 
1.7.8.4

